packages/wallet/contract/src/smartWallet.js

```diff
@@ -1,90 +1,425 @@
  * storageNode: ERef<StorageNode>,
  * zoe: ERef<ZoeService>,
  * }} shared
  */
 export const makeSmartWallet = async (
-  { address, bank, myAddressNameAdmin },
-  { agoricNames, board, namesByAddress, storageNode, zoe },
+  { address, bank },
+  { board, invitationIssuer, storageNode, zoe },
 ) => {
   assert.typeof(address, 'string', 'invalid address');
   assert(bank, 'missing bank');
-  assert(myAddressNameAdmin, 'missing myAddressNameAdmin');
+  assert(invitationIssuer, 'missing invitationIssuer');
   assert(storageNode, 'missing storageNode');
+  const invitationBrand = await E(invitationIssuer).getBrand();
+  const marshaller = await E(board).getReadonlyMarshaller();
 
-  const cacheStorageNode = E(storageNode).makeChildNode('cache');
+  // #region precious state
+  /**
+   * Private purses
+   *
+   * @type {MapStore<Brand, ERef<Purse<'set'>>>}
+   */
+  const brandPurses = makeScalarBigMapStore('brand purses', { durable: true });
 
-  const wallet = await makeWallet(bank, {
-    agoricNames,
-    cacheStorageNode,
-    namesByAddress,
-    // ??? why do we make this instead of passing the address itself?
-    myAddressNameAdmin,
-    zoe,
-    board,
+  /**
+   * Invitation makers yielded by offer results
+   *
+   * @type {MapStore<number, import('./continuing').InvitationMakers>}
+   */
+  const offerToInvitationMakers = makeScalarBigMapStore('invitation makers', {
+    durable: true,
   });
+  // #endregion
 
-  const admin = E(wallet).getAdminFacet();
-
-  /** @type {Record<string, ERef<Notifier<unknown>>>} */
-  const notifierParts = {
-    contacts: E(admin).getContactsNotifier(),
-    dapps: E(admin).getDappsNotifier(),
-    /** all current issuers */
-    issuers: E(admin).getIssuersNotifier(),
-    /** the latest offer or, in case of petname change, all offers */
-    offers: E(admin).getOffersNotifier(),
-    /** most recent payment */
-    payments: E(admin).getPaymentsNotifier(),
-    /** all current purses */
-    purses: E(admin).getPursesNotifier(),
-  };
-  const mutableState = fromEntries(keys(notifierParts).map(key => [key, []]));
-  const { subscriber, publisher } = makePublishKit();
-  publisher.publish({ ...mutableState });
-  entries(notifierParts).forEach(([key, notifier]) => {
-    void observeIteration(notifier, {
-      updateState: value =>
-        publisher.publish({
-          ...assign(mutableState, { [key]: value }),
-        }),
-    });
-  });
+  // #region other state management
 
-  const marshaller = wallet.getMarshaller();
+  // To ensure every offer ID is unique we require that each is a number greater
+  // than has ever been used. This high water mark is sufficient to track that.
+  let lastOfferId = 0;
+  /**
+   * What we've been told by the Bank or suggestIssuer
+   *
+   * @type {WeakMapStore<Brand, BrandDescriptor>} */
+  const brandDescriptors = makeScalarWeakMapStore();
+  /** @type {(brand: Brand) => ERef<Purse>} */
+  const getPurse = brand => brandPurses.get(brand);
```

Comment:
This won't always be a 1:1 mapping right?

Reply:
Why not? The design is that it makes a single purse for each brand it learns of.

Reply:
For now yes, as long as we can extend it in the future to support multiple purses of the same brand

Reply:
documenting in PR: decision made with product to not support multiple purses for same brand for MN-1.

of course an upgraded contract will be able to. at that point we'll need some way to specify in offer execution which purses to take funds from. For simplicity I think we won't require that specificity unless there are multiple purses. When there are, lack of specifier could throw or we could have a "default" purse for each brand.

Reply:
> Should this go into the source code somewhere?

yes, please. I prefer jsdoc, but other places are ok too.

Reply:
added